components/proxy/src/main/kotlin/com/hotels/styx/services/HealthCheckMonitoringService.kt

```diff
@@ -104,18 +98,18 @@ internal class HealthCheckMonitoringService(
     internal fun runChecks(application: String, objectStore: StyxObjectStore<RoutingObjectRecord>) {
         val monitoredObjects = discoverMonitoredObjects(application, objectStore)
                 .map {
-                    val tag = healthStatusTag(it.second.tags).orElse(INACTIVE_TAG)
-                    val health = objectHealthFrom(tag).orElse(null)
-                    Triple(it.first, it.second, health)
+                    val tags = it.second.tags
+                    val objectHealth = objectHealthFrom(stateTagValue(tags), healthcheckTagValue(tags))
+                    Triple(it.first, it.second, objectHealth)
                 }
                 .filter { (_, _, objectHealth) -> objectHealth != null }
 
         val pendingHealthChecks = monitoredObjects
                 .map { (name, record, objectHealth) ->
-                    healthCheck(probe, record.routingObject, objectHealth)
+                    healthCheck(probe, record.routingObject, objectHealth!!)
                             .map { newHealth -> Triple(name, objectHealth, newHealth) }
                             .doOnNext { (name, currentHealth, newHealth) ->
```

Comment:
Please collapse the separate `map` and `doOnNext` operators to a single `doOnNext`.

Reply:
I think the `map` is required to change the type of the `pendingHealthChecks` mono to the triple, which is used in the following subscriber. The `doOnNext` alone doesn't do this.

Reply:
Yeah I noticed that retrospectively